### ISSUE TITLE
How to save the r, p, f1 of plot_mc_curve in txt

### ISSUE THREAD

**Issue by larrywal-express** [ppogg/YOLOv5-Lite#11]
Thanks for your good work. I will like to save the txt file of R, P, F1.. How to incorporate this into the code in metrics.py?

**Comment by ppogg** [comment]
Hi, friend, I don't know what you mean?  Could you talk more in detail？

**Comment by larrywal-express** [comment]
After training completion, figure of Recall, Precision, and F1 are generated. Now, how to generate the txt file for Recall vs Confidence, Precision vs Confidence, and F1 vs Confidence?

**Comment by ppogg** [comment]
Actually, you can draw and save txt files by using test.py
The code to save the txt file is as follows:
```
# Append to text file
            if save_txt:
                gn = torch.tensor(shapes[si][0])[[1, 0, 1, 0]]  # normalization gain whwh
                for *xyxy, conf, cls in predn.tolist():
                    xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
                    line = (cls, *xywh, conf) if save_conf else (cls, *xywh)  # label format
                    with open(save_dir / 'labels' / (path.stem + '.txt'), 'a') as f:
                        f.write(('%g ' * len(line)).rstrip() % line + '\n')
```
 Please remember to set `python test.py --save-txt` When you use it

**Comment by larrywal-express** [comment]
not this.. i mean in the metrics.py file.. 

def plot_mc_curve(px, py, save_dir='mc_curve.png', names=(), xlabel='Confidence', ylabel='Metric'):
    # Metric-confidence curve
    fig, ax = plt.subplots(1, 1, figsize=(9, 6), tight_layout=True)

    if 0 < len(names) < 21:  # display per-class legend if < 21 classes
        for i, y in enumerate(py):
            ax.plot(px, y, linewidth=1, label=f'{names[i]}')  # plot(confidence, metric)
    else:
        ax.plot(px, py.T, linewidth=1, color='grey')  # plot(confidence, metric)

    y = py.mean(0)
    ax.plot(px, y, linewidth=3, color='blue', label=f'all classes {y.max():.2f} at {px[y.argmax()]:.3f}')
    ax.set_xlabel(xlabel)
    ax.set_ylabel(ylabel)
    ax.set_xlim(0, 1)
    ax.set_ylim(0, 1)
    plt.legend(bbox_to_anchor=(1.04, 1), loc="upper left")
    fig.savefig(Path(save_dir), dpi=250)
    


i added the line below but not working
    np.savetxt(Path(names + '.txt'), np.column_stack([px,y]))

**Comment by larrywal-express** [comment]
i mean how to obtain it in **txt file** apart from the figure plot
